app/main.py

```diff
@@ -1 +1,36 @@
-# write your code here
+from decimal import Decimal
+from json import dump, load
+
+
+def calculate_profit(trade_history):
+    sum_profit = 0
+    for trade in trade_history:
+        price = Decimal(trade["matecoin_price"])
+        if trade["bought"]:
+            sum_profit += Decimal(trade["bought"]) * price
+        if trade["sold"]:
+            sum_profit -= Decimal(trade["sold"]) * price
+    return sum_profit
+
+
+def calculate_balance_account(trade_history):
+    matecoin_account = 0
+    for item in trade_history:
+        if item["bought"]:
+            matecoin_account += Decimal(item["bought"])
+        if item["sold"]:
+            matecoin_account -= Decimal(item["sold"])
+    return matecoin_account
+
+
+if __name__ == "__main__":
+    with open("trades.json", "r") as trade_file:
```

Comment:
Read the description more meticulously. You should create the `calculate_profit` function.

Reply:
It is very good that you do everything in advance. Keep going.